### PR TITLE
Update NEST version detection for v3.4

### LINKED ISSUE #856: Upgrade NEST version detection after NEST 3.4 release
We need to upgrade the automatic version detection of NEST Simulator after the 3.4 release.

### PR DESCRIPTION
**Pull request by pnbabu**
Fixes #856 

This must be merged after NEST 3.4 is released.

### CODE REVIEW

File: .github/workflows/nestml-build.yml [hunk 1]
```diff
@@ -52,7 +52,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        nest_branch: ["v2.20.2", "v3.0", "v3.1", "v3.2", "v3.3", "master"]
+        nest_branch: ["v2.20.2", "v3.0", "v3.1", "v3.2", "v3.3", "v3.4", "master"]
```
Comment (clinssen) [inline .github/workflows/nestml-build.yml:55]: Shall we do a test run with the latest release candidate for now?
```suggestion
        nest_branch: ["v2.20.2", "v3.0", "v3.1", "v3.2", "v3.3", "v3.4_rc2", "master"]
```

**Review by clinssen — APPROVED** [review]
Cheers! Could I ask you to also review the short bit of documentation I added?

**Comment by pnbabu** [comment]
> Cheers! Could I ask you to also review the short bit of documentation I added?

The documentation looks good to me. Since I am the author of this PR, Github doesn't allow me to approve it. I think we can safely merge this as the CI is green.

**Comment by clinssen** [comment]
Thank you!

**Pull request merged**